R/measures.R

```diff
@@ -717,6 +717,133 @@ measureHAMLOSS = function(truth, response) {
   mean(truth != response)
 }
 
+#' @export subset01
+#' @rdname measures
+#' @format none
+subset01 = makeMeasure(id = "subset01", minimize = TRUE, best = 0, worst = 1,
+  properties = c("multilabel", "req.pred", "req.truth"),
+  name = "Subset-0-1 loss",
+  fun = function(task, model, pred, feats, extra.args) {
+    measureSUBSET01(getPredictionTruth.PredictionMultilabel(pred),
+    getPredictionResponse.PredictionMultilabel(pred))
+  }
+)
+
+#' @export measureSUBSET01
+#' @rdname measures
+#' @format none
+measureSUBSET01 = function(truth, response) {
+  mean(!apply(truth == response, 1, all))
+}
+
+#' @export f1mult
+#' @rdname measures
+#' @format none
+f1mult = makeMeasure(id = "f1mult", minimize = FALSE, best = 1, worst = 0,
+  properties = c("multilabel", "req.pred", "req.truth"),
+  name = "F1 measure",
+  fun = function(task, model, pred, feats, extra.args) {
+    measureF1MULT(getPredictionTruth.PredictionMultilabel(pred),
+    getPredictionResponse.PredictionMultilabel(pred))
+  }
+)
+
+#' @export measureF1MULT
+#' @rdname measures
+#' @format none
+measureF1MULT = function(truth, response) {
+  Fi = as.numeric()
```

Comment:
We know the length of `Fi` right? Then it is much faster for the for-loop, if you use the preallocation `Fi = numeric(nrow(truth))`. Do this for all measures. Never use preallocations without specifying the vector length if you can avoid this.